listenbrainz/webserver/views/metadata_api.py

```diff
@@ -295,6 +297,137 @@ def get_mbid_mapping():
     return jsonify({})
 
 
+def make_acrr_input(recording):
+    if "release_name" in recording:
+        return ArtistCreditRecordingReleaseLookupInput(
+            artist_credit_name=recording["artist_name"],
+            recording_name=recording["recording_name"],
+            release_name=recording["release_name"],
+        )
+    else:
+        return None
+
+
+def make_acr_input(recording):
+    return ArtistCreditRecordingLookupInput(
+        artist_credit_name=recording["artist_name"],
+        recording_name=recording["recording_name"],
+    )
+
+
+def make_mapping_input(recording):
+    return MBIDMappingInput(
+        artist_credit_name=recording["artist_name"],
+        recording_name=recording["recording_name"],
+    )
+
+
+def process_bulk_lookup_results(all_results, all_params, query, make_input):
+    query_params = []
+    reverse_index = {}
+    param_idx = 0
+
+    for original_idx, recording in all_params.items():
+        param = make_input(recording)
+        if param is None:
+            continue
+        query_params.append(param)
+        reverse_index[param_idx] = original_idx
+        param_idx += 1
+
+    if not query_params:
+        return all_results, all_params
+
+    query_results = query.fetch(query_params, RequestSource.json_post)
+
+    for query_result in query_results:
+        original_idx = reverse_index[query_result.index]
+        original_result = all_results[original_idx]
+
+        result_dict = query_result.dict(exclude={"index", "artist_credit_arg", "recording_name_arg", "release_name_arg"})
+        original_result.update(**result_dict)
+        all_params.pop(original_idx)
+
+    return all_results, all_params
+
+
+@metadata_bp.route("/lookup/", methods=["POST"])
+@crossdomain
+@ratelimit()
+def get_mbid_mapping_post():
+    """
+    This endpoint is the POST version for looking up recording mbids and associated MusicBrainz data. It allows up to
+    max number of items allowed. (:data:`~webserver.views.api.MAX_ITEMS_PER_GET` items)
```

Comment:
I think for this endpoint we need to lower the max number of items per call. Sending us 1,000 items to look up and then doing so in quick succession is basically a DoS attack. :(

Reply:
The aim is to support import of playlists using Troi, an alternative would be to have import logic in LB codebase instead but that would be less ideal. I have reduced the limit to 50 items per call.